staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -101,6 +101,27 @@ type Lister interface {
 	List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error)
 }
 
+// ListerWithOptions is an object that can retrieve resources that match the provided field
+// and label criteria and takes additional options on the list request.
+type ListerWithOptions interface {
+	// NewList returns an empty object that can be used with the List call.
+	// This object must be a pointer type for use with Codec.DecodeInto([]byte, runtime.Object)
+	NewList() runtime.Object
+
+	// List selects resources in the storage which match to the selector. 'options' can be nil.
+	// The extraOptions object passed to it is of the same type returned by the NewListOptions
+	// method.
+	List(ctx context.Context, options *metainternalversion.ListOptions, extraOptions runtime.Object) (runtime.Object, error)
+
+	// NewListOptions returns an empty options object that will be used to pass extra options
+	// to the List method. It may return a bool and a string, if true, the
+	// value of the request path below the list will be included as the named
+	// string in the serialization of the runtime object. E.g., returning "path"
+	// will convert the trailing request scheme value to "path" in the map[string][]string
+	// passed to the converter.
+	NewListOptions() (runtime.Object, bool, string)
```

Comment:
I realize this doc is taken verbatim from `GetterOptions`, but I'm finding it difficult to understand. It feels like there is implicit context that should be explained, and given `GetterOptions` was introduced in 2015, it would be worth a refresh of it to help future maintainers?

Reply:
I agree. I favoured consistency, though we can take that opportunity to refresh wording and provide more context.